### PR TITLE
feat: Send CAR file, and use codecs

### PR DESCRIPTION
**Pull request by ukstv**
Treat that as a work in progress for now.

Ah, it supposedly fixes AWS S3 initialisation bug.

### CODE REVIEW

File: src/services/request-presentation-service.ts [hunk 1]
```diff
@@ -1,99 +1,135 @@
...
-   */
-  async body(request: Request): Promise<RequestPresentation> {
+  constructor(
+    private readonly anchorRepository: IAnchorRepository,
+    private readonly merkleCarService: IMerkleCarService,
+    private readonly witnessService: WitnessService
+  ) {}
+
+  async witnessCAR(anchor: AnchorWithRequest | null): Promise<CAR | null> {
+    if (!anchor) return null // TODO Add metric
```
Comment (stbrody) [inline src/services/request-presentation-service.ts:46]: can we file a ticket for this todo and reference the ticket number here?

File: src/services/request-presentation-service.ts [hunk 2]
```diff
@@ -1,99 +1,135 @@
...
+    const cacheKey = anchor.cid.toString()
+    const fromCache = this.cache.get(cacheKey)
+    if (fromCache) return fromCache
+    const merkleCAR = await this.merkleCarService.retrieveCarFile(anchor.proofCid)
+    if (!merkleCAR) return null // TODO Add metric
```
Comment (stbrody) [inline src/services/request-presentation-service.ts:51]: we should probably log a warning for this case as it isn't expected

File: src/services/request-presentation-service.ts [hunk 3]
```diff
@@ -1,99 +1,135 @@
...
     switch (status) {
       case RequestStatus.COMPLETED: {
         const anchor = await this.anchorRepository.findByRequest(request)
+        const witnessCAR = await this.witnessCAR(anchor)
         // TODO: This is a workaround, fix in CDB-2192
-        const anchorCommit = {
-          cid: anchor ? anchor.cid.toString() : request.cid,
+        const anchorCommit: CommitPresentation = {
+          cid: anchor ? anchor.cid : CID.parse(request.cid),
           content: {
```
Comment (stbrody) [inline src/services/request-presentation-service.ts:66]: not for this PR, but as a general question - should we remove this content object from the response entirely if it's not being used anywhere?

**Review by stbrody — APPROVED** [review]
LGTM!

File: src/services/request-presentation-service.ts (hunk 1 above)
Comment (ukstv) [reply src/services/request-presentation-service.ts]: Better do that here :D

File: src/services/request-presentation-service.ts (hunk 2 above)
Comment (ukstv) [reply src/services/request-presentation-service.ts]:
```ts
logger.warn(`No Merkle CAR found for anchor ${anchor.cid}`)
```

File: src/services/request-presentation-service.ts (hunk 3 above)
Comment (ukstv) [reply src/services/request-presentation-service.ts]: Yep. That was done in js-ceramic codec. And, removed `content` from CAS codebase as well.

**Pull request merged**